src/pipeline.ts

```diff
@@ -816,6 +938,68 @@ function snakeCaseKeys<T = unknown>(obj: T, sep = '-'): T {
   return result as any;
 }
 
+/**
+ * Multiple stages that are deployed in parallel
+ *
+ * A `Wave`, but with addition GitHub options
+ *
+ * Create with `GitHubWorkflow.addWave()` or `GitHubWorkflow.addGitHubWave()`, do not construct directly
+ *
+ * @hideconstructor
+ */
+export class GitHubWave extends Wave {
+  constructor(
+    /** Identifier for this Wave */
+    public readonly id: string,
+    /** GitHubWorkflow that this wave is part of  */
+    private pipeline: GitHubWorkflow,
+    props: WaveProps = {},
+  ) {
+    super(id, props);
+  }
+
+  /**
+   * Add a Stage to this wave
+   *
+   * It will be deployed in parallel with all other stages in this
+   * wave.
+   */
+  public addStage(stage: Stage, options: AddStageOpts = {}) {
+    const stageDeployment = super.addStage(stage, options);
+    this.pipeline.addingStageFromWave(stage, stageDeployment);
+    return stageDeployment;
+  }
+
+  /**
+   * Add a Stage to this wave
+   *
+   * It will be deployed in parallel with all other stages in this
+   * wave.
+   */
+  public addStageWithGitHubOptions(
+    stage: Stage,
+    options?: AddGitHubStageOptions,
+  ): StageDeployment {
+    const stageDeployment = super.addStage(stage, options);
+    this.pipeline.addingStageFromWave(stage, stageDeployment, options);
+    return stageDeployment;
+  }
+
+  // /**
+  //  * Add an additional step to run before any of the stages in this wave
+  //  */
+  // public addPre(...steps: Step[]) {
+  //   this.pre.push(...steps);
+  // }
+
+  // /**
+  //  * Add an additional step to run after all of the stages in this wave
+  //  */
+  // public addPost(...steps: Step[]) {
+  //   this.post.push(...steps);
+  // }
```

Comment:
oops?

Reply:
Well, I kept these here commented for a few reasons that probably don't hold water:

1. If we did decide to override them the signature is there
2. To indicate that they are there and still usable, but not currently overridden
3. I was sure that some test case would show that I needed to add some logic there. It never did happen.

In any case, I'll remove these. They're easy to restore.